### ISSUE TITLE
Unexpected behavior in Births due to truncated rather than probabilistic equation

### ISSUE THREAD

**Issue by edwenger** [starsimhub/starsim#615]
https://github.com/starsimhub/starsim/blob/main/starsim/demographics.py#L108

`n_new = int(np.floor(sim.people.alive.count() * scaled_birth_prob))`

Reproduce with simple example by toggling `rel_birth` between 33.33 and 33.34 (noting that default CBR is 30 per 1000)

```
import starsim as ss

people = ss.People(n_agents=1)

sim = ss.Sim(
    people=people, 
    diseases=[], networks=[], 
    demographics=[ss.Births(rel_birth=33.34))],
    n_years=5)

sim.run()
sim.plot()
```

I suppose the expected behavior would be a binomial process, but there may be some reason I'm unaware of (e.g. common random number handling) that this isn't implemented here currently.

**Comment by daniel-klein** [comment]
Note that remnants of previous discussions about moving away from this deterministic approach to births are discussed in #404.

**Comment by cliffckerr** [comment]
Closed by #724
